mod/state-transition/pkg/core/deposits_validation.go

```
@@ -0,0 +1,140 @@
+		if _, err := st.GetEth1DepositIndex(); err == nil {
+			// there should not be Eth1DepositIndex stored before
+			// genesis first deposit
+			return ErrDepositMismatch
+		}
+		if len(deposits) == 0 {
+			// there should be at least a validator in genesis
+			return ErrDepositsLengthMismatch
+		}
+		for i, deposit := range deposits {
+			// deposits indexes should be contiguous
+			if deposit.GetIndex() != math.U64(i) {
```

golangci-lint (1.62.2)

[high] G115: integer overflow conversion int -> uint64

(gosec)